src/js/user.js

```diff
@@ -51,15 +85,31 @@ function UserManager(auth, server) {
                         return _.find(tokens, tokenStub);
                     })
                     .then(function (token) {
-                        return token ? token : api.createToken(tokenStub, userObj);
+                        // delete existing token on server
+                        if (token) {
+                            let usr = { 
+                                username: userObj.username, 
+                                password: userObj.password, 
+                                tokenId: token.id
+                            }
+                            return deleteAccessToken(usr)
+                                    .then(() => api.createToken(tokenStub, userObj));
+                        } else {
+                            return api.createToken(tokenStub, userObj);
+                        }
                     })
-                    .then(function (token) {
+                    .then(function (token) { 
+                        user.tokenId = token.id;
                         user.token = token.sha1;
                         return user;
                     });
             });
         },
 
+        logout: function(user) {
+            deleteAccessToken(user);
```

Comment:
I think we can handle the return of this promise. I don't think it's good practice to leave a promise hanging like this.

It could be as simple as
`        logout: function(user) {
            deleteAccessToken(user)
                .then(res => console.log(res))
                .catch(err => console.log(err));
        },`

but I think we should probably handle it.

Reply:
nice catch for the return! 
In the case there's an error, it will log to the dev console anyway, so we don't need to do that manually.

Reply:
204 is successful